Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -533,18 +533,32 @@ variable [RingHomCompTriple σ₁₂ σ₂₃ σ₁₃]
 variable (f : M₂ →ₛₗ[σ₂₃] M₃) (g : M₁ →ₛₗ[σ₁₂] M₂)
 
 /-- Composition of two linear maps is a linear map -/
+@[pp_dot]
 def comp : M₁ →ₛₗ[σ₁₃] M₃ where
   toFun := f ∘ g
   map_add' := by simp only [map_add, forall_const, Function.comp_apply]
   map_smul' r x := by simp only [Function.comp_apply, map_smulₛₗ, RingHomCompTriple.comp_apply]
 #align linear_map.comp LinearMap.comp
 
-set_option quotPrecheck false in -- Porting note: error message suggested to do this
 /-- `∘ₗ` is notation for composition of two linear (not semilinear!) maps into a linear map.
 This is useful when Lean is struggling to infer the `RingHomCompTriple` instance. -/
 infixr:80 " ∘ₗ " =>
-  @LinearMap.comp _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ (RingHom.id _) (RingHom.id _) (RingHom.id _)
-    RingHomCompTriple.ids
+  LinearMap.comp (σ₁₂ := RingHom.id _) (σ₂₃ := RingHom.id _) (σ₁₃ := RingHom.id _)
+
+open Qq Lean PrettyPrinter.Delaborator SubExpr in
+/-- Use the `∘ₗ` notation in the infoview when possible. -/
+@[delab app.LinearMap.comp]
+def delabCompLinear : Delab := do
+  let e  ← getExpr
+  -- it would be really nice to use `~q()` here
+  guard <| e.isAppOfArity ``LinearMap.comp 21
+  -- check the map is linear, not semilinear
+  let `(RingHom.id $_) ← withNaryArg 15 delab | failure
+  let `(RingHom.id $_) ← withNaryArg 16 delab | failure
+  let `(RingHom.id $_) ← withNaryArg 17 delab | failure
+  let f ← withNaryArg 19 delab
+  let g ← withNaryArg 20 delab
+  `($f ∘ₗ $g)
```

Comment:
We should be sure to make this controllable using `pp.notation`. For efficiency, we can also avoid delaborating the three arguments and instead check that the expression is literally `RingHom.id` applied to two arguments, though this will have different behavior depending on whether these arguments are somehow in another form that just pretty prints as `RingHom.id _`.
```suggestion
def delabCompLinear : Delab := whenPPOption getPPNotation do
  let e ← getExpr
  let numArgs := 21
  guard <| e.isAppOfArity ``LinearMap.comp numArgs
  guard <| (e.getArg! 15 (n := numArgs)).isAppOfArity ``RingHom.id 2
  guard <| (e.getArg! 16 (n := numArgs)).isAppOfArity ``RingHom.id 2
  guard <| (e.getArg! 17 (n := numArgs)).isAppOfArity ``RingHom.id 2
  let f ← withNaryArg 19 delab
  let g ← withNaryArg 20 delab
  `($f ∘ₗ $g)
```

I've been meaning to make a pretty printer command that lets you generate matchers. 

There's a PR that should be landing soon that will at least make it so that you can replace `infixr` with
```
notation3:80 f:81 " ∘ₗ " g:80 =>
  LinearMap.comp (σ₁₂ := RingHom.id _) (σ₂₃ := RingHom.id _) (σ₁₃ := RingHom.id _) f g
```
and it *should* be able to generate a pretty printer automatically.

Reply:
That PR I mentioned was just merged yesterday. Want to check if `notation3` works here and actually pretty prints?

Reply:
Done in #8847